core/src/main/java/org/infinispan/container/entries/VersionedRepeatableReadEntry.java

```diff
@@ -102,6 +103,9 @@ private boolean skewed(IncrementableEntryVersion prevVersion, EntryVersion versi
          }
          IncrementableEntryVersion prevVersion = versionFromEntry(ice);
          if (prevVersion == null) {
+            if (rollingUpgrade) {
+               return versionGenerator.nonExistingVersion();
```

Comment:
Wouldn't it be better to return `nonExistingVersion()` every time, without adding the `rollingUpgrade` parameter?

Reply:
I'm not sure, wdyt @pruivo ?

Reply:
we never have `prevVersion==null` except for the rolling upgrade scenario. 
Hot Rod isn't aware of the internal version and Infinispan stores the entry from RemoteStore as is. 
I would leave the exception there at least to make sure there are no errors.